page_separator.py

```diff
@@ -1,11 +1,27 @@
-from PyPDF2 import PdfFileWriter, PdfFileReader
 import argparse
+import sys
+from enum import Enum
+
+from PyPDF2 import PdfFileWriter, PdfFileReader
 from PyPDF2.pdf import PageObject
 
-def is_letter(page: PageObject) -> bool:
-    if page.mediaBox.getHeight() == 792:
-        return True
-    return False
+LETTER_RATIO = 11 / 8.5
+LEGAL_RATIO = 14 / 8.5
+MARGIN_OF_ERROR = 0.05 # page ratio should differ by at most this percent

```

Comment:
Maybe leave a comment for future reference that the MARGIN_OF_ERROR may not exceed 3/25.

Reply:
why not? cuz then it'll overlap?

Reply:
That is correct. overlap in itself is fine, but the code will bias towards categorizing the page as letter.

Reply:
Even higher ratio than that will make the code to incorrectly categorize the page, so it's probably safer to limit it.